### PR TITLE
PublicDashboard -> PublicOrEmbeddedDashboard

### LINKED ISSUE #43289: rename `PublicDashboard` to `PublicOrEmbeddedDashboard` and isPublic to isPublicOrEmbed

### PR DESCRIPTION
**Pull request by npretto**
Closes https://github.com/metabase/metabase/issues/43289, part of the preparation work for https://github.com/metabase/metabase/issues/42628

"Public" is a bit confusing as when we talk about public sharing we mean public links, but the code uses Public also for static embedded related things.

We [discussed about it](https://metaboat.slack.com/archives/C063Q3F1HPF/p1716821967657249) and decided to make the name more explicit.

The pr should be small enough to be readable on its own, but I tried to make commits pretty small anyway in case you want to read it commit by commit

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx [hunk 1]
```diff
@@ -62,7 +62,8 @@ export interface DashCardProps {
   isFullscreen?: boolean;
   isMobile?: boolean;
   isNightMode?: boolean;
-  isPublic?: boolean;
+  /** If public sharing or static/public embed */
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx:65]: do we really need to duplicate this comment to every place this flag exists? Does that mean the name itself isn't clear enough? I don't have a strong conviction if we should keep or remove them. 5 of them don't bother me much. But it just sparks this question.

File: frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { PublicOrEmbeddedQuestion as PublicOrEmbeddedQuestion } from "./PublicOrEmbeddedQuestion";
```
Comment (WiNloSt) [inline frontend/src/metabase/public/containers/PublicOrEmbeddedQuestion/index.ts:1]: nit: I know this is from using F2 to rename in VSCode. I get this all the time 😆 

```suggestion
export { PublicOrEmbeddedQuestion } from "./PublicOrEmbeddedQuestion";
```

**Review by WiNloSt — APPROVED** [review]
Looks good to me with very minor comments. 🙏

File: frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx (hunk 1 above)
Comment (npretto) [reply frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx:65]: I don't love it, but I didn't come up with a better alternative, I know copy pasted things are not perfect but putting the comment only in one place wasn't perfect either.
This makes me wonder if this prop should be a flag on redux 🙃, but that's a bit to invasive of a refactor for now

Comment (WiNloSt) [reply frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx:65]: you have a good point about centralizing the prop, but not for this PR. Let's leave it as is for now.

**Review by oisincoveney — APPROVED** [review]

**Pull request merged**